docs/articles/documentation/test-api/selecting-page-elements/examples-of-working-with-page-dom.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: docs
+title: Examples of Working with Page DOM
+permalink: /documentation/test-api/selecting-page-elements/examples-of-working-with-page-dom.html
+---
+# Examples of Working with Page DOM
+
+This document shows how to work with page DOM in frequent real-world situations.
+
+* [Accessing Page Element Properties](#accessing-page-element-properties)
+* [Getting a Page Element Using Custom Logic](#getting-a-page-element-using-custom-logic)
+* [Checking if an Element is Available](#checking-if-an-element-is-available)
+* [Enumerating Elements Identified by a Selector](#enumerating-elements-identified-by-a-selector)
+
+## Accessing Page Element Properties
+
+To work with page elements, use TestCafe [selectors](selectors/README.md).
+Import the `Selector` function and call it passing a CSS selector inside.
+This function creates a selector object [whose API](dom-node-state.md) exposes the most used members of HTML element API.
+
+Note that all property getters are asynchronous, so add the `await` keyword.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const element     = Selector('#developer-name');
+    const clientWidth = await element.clientWidth;
+
+    // do something wuth clientWidth
+});
+```
+
+If you need to access an element property not included in the selector's API, request it explicitly by using the [addCustomDOMProperties](selectors/extending-selectors.md) method.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    });
+
+    await t.expect(label.innerHTML).contains('input type="checkbox" name="remote"');
+});
+```
+
+## Getting a Page Element Using Custom Logic
+
+Sometimes CSS selectors are not powerful enough to identify the required page element.
+In this instance, you can introduce a function that picks the desired element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const rCheckBoxes = Selector(() => {
```

Comment:
`rCheckBoxes ` is a bad name for variable, What does `r` mean?

Reply:
oops, I don't remember why I called it this way

Reply:
Ah, because they start with 'R'

Reply:
Let's choose a clear name.